builtInFunctions/esdtLocalMint.go

```diff
@@ -107,6 +106,14 @@ func (e *esdtLocalMint) ProcessBuiltinFunction(
 	return vmOutput, nil
 }
 
+func (e *esdtLocalMint) isAllowedToMint(acntSnd vmcommon.UserAccountHandler, tokenID []byte) error {
+	if e.crossChainTokenChecker.IsCrossChainOperation(tokenID) && e.crossChainTokenChecker.IsSelfMainChain() {
```

Comment:
we do not need IsSelfMainChain here. We actually need a whitelist check. Only addresses from the whitelist can mint - those addresses are the built in cross chain op contracts in case of the mainchain and in case of the sovereign, those will be empty - as there is no need for mint in mainchain to sovereign cross chain process.

Also we need this for ESDTNFTCreate and ESDTNFTAddQuantity as well. The same with whitelist.

crossChainToken checker could have this whitelist map of addresses.